packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/lambda.js

```diff
@@ -0,0 +1,118 @@
+  propagateInstanaHeaders(originalArgs, span, suppressed = false) {
+    const params = originalArgs[0].input;
+    let clientContextContentBase64;
+    let clientContextContentJSON;
+    let isJSON = true;
+
+    const instanaHeaders = {
+      'x-instana-l': suppressed ? '0' : '1'
+    };
+
+    if (span) {
+      instanaHeaders['x-instana-s'] = span.s;
+      instanaHeaders['x-instana-t'] = span.t;
+    }
+    const operation = this.convertOperationName(originalArgs[0].constructor.name);
+    // invokeAsync doesn't have the option ClientContext
+    if (operation === 'invoke') {
+      if (params && params.ClientContext != null) {
+        clientContextContentBase64 = Buffer.from(params.ClientContext, 'base64').toString();
+
+        try {
+          clientContextContentJSON = JSON.parse(clientContextContentBase64);
+          if (typeof clientContextContentJSON.Custom === 'object') {
+            Object.assign(clientContextContentJSON.Custom, instanaHeaders);
+          } else {
+            clientContextContentJSON.Custom = instanaHeaders;
+          }
+        } catch (err) {
+          // ClientContext has a value that is not JSON, then we cannot add Instana headers
+          isJSON = false;
+        }
+      } else {
+        clientContextContentJSON = {
+          Custom: instanaHeaders
+        };
+      }
+
+      if (isJSON) {
+        clientContextContentBase64 = Buffer.from(JSON.stringify(clientContextContentJSON), 'utf8').toString('base64');
+
+        if (clientContextContentBase64.length <= MAX_CONTEXT_SIZE) {
+          params.ClientContext = clientContextContentBase64;
+        }
+      }
+    }
+  }
+
+  instrumentedSmithySend(ctx, isActive, originalSend, smithySendArgs) {
+    const self = this;
+    const skipTracingResult = cls.skipExitTracing({ isActive, extendedResponse: true });
+    if (skipTracingResult.skip) {
+      if (skipTracingResult.suppressed) {
+        this.propagateInstanaHeaders(smithySendArgs, null, skipTracingResult.suppressed);
+      }
+
+      return originalSend.apply(ctx, smithySendArgs);
+    }
+
+    return cls.ns.runAndReturn(() => {
+      const span = cls.startSpan(SPAN_NAME, EXIT);
+      span.ts = Date.now();
+      span.stack = tracingUtil.getStackTrace(this.instrumentedMakeRequest, 1);
+      span.data[this.spanName] = this.buildSpanData(smithySendArgs[0], smithySendArgs[0].input);
+
+      this.propagateInstanaHeaders(smithySendArgs, span);
+
+      const { originalCallback, callbackIndex } = tracingUtil.findCallback(smithySendArgs);
+      if (callbackIndex !== -1) {
+        smithySendArgs[callbackIndex] = cls.ns.bind(function (err) {
+          self.finishSpan(err, span);
+          return originalCallback.apply(this, arguments);
+        });
+        return originalSend.apply(ctx, smithySendArgs);
+      } else {
+        const request = originalSend.apply(ctx, smithySendArgs);
+        request
+          .then(() => {
+                        this.finishSpan(null, span);
```

Comment:
This looks like a wrong indention. Wondering how that is possible 🤔

Reply:
Ah right, Thanks for pointing this, I will update this